dali/pipeline/operator/builtin/external_source.h

```diff
@@ -221,7 +223,15 @@ class ExternalSource : public Operator<Backend>, virtual public BatchSizeProvide
         device_id_(spec.GetArgument<int>("device_id")),
         dtype_(spec.GetArgument<DALIDataType>("dtype")),
         previous_dtype_(DALIDataType::DALI_NO_TYPE),
+        ndim_(-1),
+        layout_(),
         sync_worker_(device_id_, false) {
+    if (spec.TryGetArgument(ndim_, "ndim")) {
+      DALI_ENFORCE(ndim_ >= 0, "Incorrect number of dimensions. "
+                   "Use positive values for tensors or 0 for scalars.");
+    }
+    layout_ = spec.GetArgument<TensorLayout>("layout");
```

Comment:
I'd still suggest:
```suggestion
    spec.TryGetArgument(layout_, "layout");
```
and get rid of the default value. The result will be the same, but the documentation will be cleaner.